File: utils/trie/triemap/triemap.go
package trieMap

type TrieMap[E any] interface {
	Put(key string, value E)
	Get(key string) E
	Remove(key string)
	ContainsKey(key string) bool
	ShortestPrefixOf(query string) string
	LongestPrefixOf(query string) string
	KeysWithPrefix(prefix string) []string
	HasKeyWithPrefix(prefix string) bool
	KeysWithPattern(pattern string) []string
	HasKeyWithPattern(pattern string) bool
	Len() int
}

File: 648. Replace Words/src/TrieMap.java
import java.util.List;

public interface TrieMap<E> {
    public void put(String key, E val);
    public void remove(String key);
    public E get(String key);
    public boolean containsKey(String key);
    public String shortestPrefixOf(String query);
    public String longestPrefixOf(String query);
    public List<String> keysWithPrefix(String prefix);
    public boolean hasKeyWithPrefix(String prefix);
    public List<String> keysWithPattern(String pattern);
    public boolean hasKeyWithPattern(String pattern);
    public int size();
}
